### ISSUE TITLE
Geopy adapter objects are not pickleable

### ISSUE THREAD

**Issue by dchigarev** [geopy/geopy#449]
Geopy adapters are not pickleable, however it seems that all of them are just wrappers for adapters from `requests` library (which supports pickling), so geopy adapters should also support pickling I guess.

**Code to reproduce:**
```python
import geopy
import pickle

adapter = geopy.geocoders.Yandex(api_key="random_key").adapter

adapter_bin = pickle.dumps(adapter)
adapter_ser = pickle.loads(adapter_bin)  # AttributeError
```

<details><summary>Traceback</summary>

```
Exception ignored in: <function RequestsAdapter.__del__ at 0x000002976DAB73A8>
Traceback (most recent call last):
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\lib\site-packages\geopy\adapters.py", line 366, in __del__
    self.session.close()
AttributeError: 'RequestsAdapter' object has no attribute 'session'
Traceback (most recent call last):
  File "C:\Users\dchigare\Desktop\REPOS\TESTS\repr3.py", line 8, in <module>
    adapter_ser = pickle.loads(adapter_bin)
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\lib\site-packages\requests\adapters.py", line 144, in __setstate__
    block=self._pool_block)
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\lib\site-packages\geopy\adapters.py", line 534, in init_poolmanager
    if self.__ssl_context is not None:
AttributeError: 'RequestsHTTPWithSSLContextAdapter' object has no attribute '_RequestsHTTPWithSSLContextAdapter__ssl_context'
```

</details>

In objects from `requests` library, we often can see that pattern of supporting pickling:

- [object has `__attr__` field](https://github.com/psf/requests/blob/143150233162d609330941ec2aacde5ed4caa510/requests/adapters.py#L110-L111), that stores available attributes
- [object has `__getstate__` method](https://github.com/psf/requests/blob/143150233162d609330941ec2aacde5ed4caa510/requests/adapters.py#L131-L132), that returns us dictionary with attributes from `__attr__`
- [object has `__setstate__` method](https://github.com/psf/requests/blob/143150233162d609330941ec2aacde5ed4caa510/requests/adapters.py#L134-L144), that sets the `__attr__` atributes

So, when geopy adapters inherits from `requests` adapters they often add new fields for the class, but do not add them into `__attr__` list, and I think that's the root cause of the problem.

**Comment by KostyaEsmukov** [comment]
Why would you ever want to pickle adapters? Could you please give a practical example where this could be useful?

> it seems that all of them are just wrappers for adapters from requests library

No, there are adapters for other packages, like urllib and aiohttp. See https://geopy.readthedocs.io/en/stable/#module-geopy.adapters

**Comment by KostyaEsmukov** [comment]
Closing, as I believe that adapters/HTTP sessions must not be pickleable, as they carry a transient state which cannot be restored (such as socket descriptors).

I looked at the linked issue, and I guess the reason why you asked for that is to allow transferring a geocoder instance via pickle to other processes/machines. This shouldn't be done as it is very error-prone, and a new instance should be created in each address space instead. For example, different machines might have different system proxy settings, or have different CA stores, which would result in a different restored geocoder state.
